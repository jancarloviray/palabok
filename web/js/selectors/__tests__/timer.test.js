/* @flow */
/* eslint-env jest */

/*::
  import type { State } from '../../selectors/state'
*/

import Timer from '../timer'

test('work in progress', () => {
  const state /*: State */ = {
    time: { now: new Date('2010-04-23T08:00:00Z') },
    timer: {
      active: true,
      startedAt: new Date('2010-04-23T08:00:00Z'),
      endsAt: new Date('2010-04-23T08:30:00Z'),
      lastLap: new Date('2010-04-23T08:00:00Z'),
      duration: 30 * 60000,
      laps: 0,
      labelId: '_default',
      type: 'work'
    },
    labels: {
      '_default': {
        id: '_default',
        name: 'Work',
        color: 'dodgerblue'
      }
    }
  }

  const timer = Timer.full(state)

  expect(timer).toMatchSnapshot()
})

test('work in laps', () => {
  const state /*: State */ = {
    time: { now: new Date('2010-04-23T09:00:00Z') },
    timer: {
      active: true,
      startedAt: new Date('2010-04-23T08:00:00Z'),
      endsAt: new Date('2010-04-23T08:30:00Z'),
      lastLap: new Date('2010-04-23T08:30:00Z'),
      duration: 30 * 60000,
      laps: 2,
      labelId: '_default',
      type: 'work'
    },
    labels: {
      '_default': {
        id: '_default',
        name: 'Work',
        color: 'dodgerblue'
      }
    }
  }

  const timer = Timer.full(state)

  expect(timer).toMatchSnapshot()
})

test('inactive', () => {
  const timer = Timer.full({
    time: { now: new Date('2010-04-23T08:00:00Z') },
    timer: {
      active: false
    }
  })

  expect(timer).toMatchSnapshot()
})
