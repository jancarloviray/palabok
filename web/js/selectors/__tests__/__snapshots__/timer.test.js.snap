// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inactive 1`] = `
Object {
  "active": false,
}
`;

exports[`work in laps 1`] = `
Object {
  "active": true,
  "duration": 1800000,
  "elapsed": 3600000,
  "endsAt": 2010-04-23T08:30:00.000Z,
  "isOvertime": true,
  "labelId": "_default",
  "labelText": "Work",
  "laps": 2,
  "lastLap": 2010-04-23T08:30:00.000Z,
  "progress": 1,
  "remaining": -1800000,
  "startedAt": 2010-04-23T08:00:00.000Z,
  "type": "work",
}
`;

exports[`work in progress 1`] = `
Object {
  "active": true,
  "duration": 1800000,
  "elapsed": 0,
  "endsAt": 2010-04-23T08:30:00.000Z,
  "isOvertime": 0,
  "labelId": "_default",
  "labelText": "Work",
  "laps": 0,
  "lastLap": 2010-04-23T08:00:00.000Z,
  "progress": 0,
  "remaining": 1800000,
  "startedAt": 2010-04-23T08:00:00.000Z,
  "type": "work",
}
`;
